ui/src/views/Clients/ViewClients/index.jsx

```diff
@@ -36,51 +41,35 @@ import ErrorPanel from '../../../components/ErrorPanel';
 }))
 export default class ViewClients extends PureComponent {
   state = {
-    clientSearch: '',
-    // eslint-disable-next-line react/no-unused-state
-    previousClientId: '',
-    // This needs to be initially null in order for the defaultValue to work
-    value: null,
+    search: '',
   };
 
-  static getDerivedStateFromProps(props, state) {
-    // Any time the current user changes,
-    // Reset state to reflect new user / log out and default clientSearch query
-    if (
-      props.user &&
-      props.user.credentials.clientId !== state.previousClientId
-    ) {
-      return {
-        clientSearch: props.user.credentials.clientId,
-        previousClientId: props.user.credentials.clientId,
-      };
-    }
-
-    if (!props.user && state.previousClientId !== '') {
-      return {
-        clientSearch: '',
-        previousClientId: '',
-      };
-    }
-
-    return null;
-  }
-
-  handleClientSearchSubmit = clientSearch => {
+  handleClientSearchSubmit = async search => {
     const {
       data: { refetch },
     } = this.props;
+    const searchUri = this.props.history.location.search
+      ? decodeURIComponent(
+          parse(this.props.history.location.search.slice(1)).search
+        )
+      : '';
 
-    this.setState({ clientSearch });
+    this.setState({ search });
 
-    refetch({
+    await refetch({
       clientOptions: {
-        ...(clientSearch ? { prefix: clientSearch } : null),
+        ...(search ? { prefix: search } : null),
       },
       clientsConnection: {
         limit: VIEW_CLIENTS_PAGE_SIZE,
       },
     });
+
+    if (search !== searchUri) {
+      this.props.history.push(
+        search.length > 0 ? `?${stringify({ search })}` : '/auth/clients'
+      );
```

Comment:
Would it be possible to change the url prior to `refetch`ing?

Reply:
Unfortunately, changing the URL prior to awaiting the refetch will cause loading to be stuck again.